type_templates/gltj.js

```diff
@@ -11,40 +11,49 @@ export default e => {
   app.appType = 'gltj';
 
   const srcUrl = '${this.srcUrl}';
-  
+
   let _update = null;
   e.waitUntil((async () => {
-    const res = await fetch(srcUrl);
-    const s = await res.text();
-    const j = JSON6.parse(s);
-    
-    const material = new THREE.ShaderMaterial(j);
-    
-    const mesh = new THREE.Mesh(geometry, material);
-    mesh.frustumCulled = false;
-    app.add(mesh);
-    
-    let now = 0;
-    _update = timeDiff => {
-      if (material.uniforms.iTime) {
-        material.uniforms.iTime.value = now/1000;
-        material.uniforms.iTime.needsUpdate = true;
+      const res = await fetch(srcUrl);
+      const s = await res.text();
+      const j = JSON6.parse(s);
+
+      const material = new THREE.ShaderMaterial(j);
+
+      const mesh = new THREE.Mesh(geometry, material);
+      mesh.frustumCulled = false;
+      app.add(mesh);
+
+      const uniforms = app.getComponent("uniforms");
+      for (const name in uniforms) {
+        material.uniforms[name].value = uniforms[name];
       }
-      if (material.uniforms.iResolution) {
-        if (!material.uniforms.iResolution.value) {
-          material.uniforms.iResolution.value = new THREE.Vector2();
+
+      let now = 0;
+      _update = (timestamp, timeDiff) => {
+        if (material.uniforms.iTime) {
+          material.uniforms.iTime.value = now/1000;
+          material.uniforms.iTime.needsUpdate = true;
         }
-        renderer.getSize(material.uniforms.iResolution.value)
-          .multiplyScalar(renderer.getPixelRatio());
-        material.uniforms.iResolution.needsUpdate = true;
-      }
-      
-      now += timeDiff;
-    };
-  })());
+        if(material.uniforms.iTimeS) {
+          material.uniforms.iTimeS.value = timestamp/1000;
+          material.uniforms.iTimeS.needsUpdate = true;
+        }
+        if (material.uniforms.iResolution) {
+          if (!material.uniforms.iResolution.value) {
+            material.uniforms.iResolution.value = new THREE.Vector2();

```

Comment:
It's not good to allocate in the render loop.

This could be `material.uniforms.iResolution.value.set(0,0)`